service/controller/resource/tccp/create.go

```diff
@@ -147,7 +150,7 @@ func (r *Resource) createStack(ctx context.Context, cr v1alpha1.Cluster) error {
 	{
 		r.logger.LogCtx(ctx, "level", "debug", "message", "computing the template of the tenant cluster's control plane cloud formation stack")
 
-		params, err := r.newTemplateParams(ctx, cr, time.Now())
+		params, err := newTemplateParams(ctx, cr, time.Now(), r)
```

Comment:
That interface change does not make sense. We should either have a function or a method on the resource. Giving the resource to a function is backwards. The primitive here is to simply have the method on the resource.